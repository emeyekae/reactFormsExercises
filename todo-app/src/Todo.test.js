import React from "react";
import { render, screen, fireEvent } from '@testing-library/react';
import Todo from './Todo';

//smoke test
test('renders without crashing', function() {
  render(<Todo />);
});

//snapshot test
test("matches snapshot", function() {
  const {asFragment} = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

test("run delete function", function() {
    const removeMock = jest.fn();
    const { getByText } = render(<Todo remove={removeMock} />);
    const deleteButton = getByText("X");
    fireEvent.click(deleteButton);
    expect(removeMock).toHaveBeenCalled();
  });
  
  test("match snapshot while editing", function() {
    const { asFragment, getByText } = render(<Todo />);
    const editButton = getByText("Edit");
    fireEvent.click(editButton);
    expect(asFragment()).toMatchSnapshot();
  });


test("runs update on form submit", function() {
    const updateMock = jest.fn();
    const { getByText } = render(<Todo update={updateMock} />);
    const editButton = getByText("Edit");
    fireEvent.click(editButton);
    const updateButton = getByText("Update");
    fireEvent.click(updateButton);
    expect(updateMock).toHaveBeenCalled();
  });