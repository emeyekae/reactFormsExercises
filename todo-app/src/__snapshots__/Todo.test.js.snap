// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot while editing 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        type="text"
        value="default todo"
      />
      <button>
        Update
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <li>
      default todo
    </li>
    <button>
      Edit
    </button>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
